### PR TITLE
Make initial rendering synchronous, fixes #41

### LINKED ISSUE #41: Using fastForEach makes ko.applyBindings asynchronous
I currently struggled a bit with this. I'm trying to do something like this.

```
var $node = $("<div data-bind='fastForEach: items'/>");
ko.applyBindings({ items: [1, 2, 3, 4] }, $node.get(0));
var height = $node.height(); // <-- here it reports 0 because nodes are not rendered yet
```

Maybe this is acceptable, but quite tricky to work around in some situations, and I don't like the fact that the DOM is not ready right after `applyBindings`, and I think most developers might get confused if they meet such scenario. As far as I know, this kind of synchronous behavior is present using every other binding, but correct me if I'm wrong.

Maybe it would be good idea to process the initial rendering synchronously. The main point of being asynchronous is to be able to optimize array **changes**, so it should fit into the design I think.

What do you think @brianmhunt about this?

PS: It would also eliminate the frustrating stuttering effect in the browser when applying bindings with a more complex model.

### PR DESCRIPTION
**Pull request by cervengoc**
This PR makes processing of initial data synchronous, so for example `ko.applyBindings` will be synchronous too, and therefore the bound nodes will be ready after calling it.

Fixes #41

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -166,7 +166,10 @@ FastForEach.prototype.onArrayChange = function (changeSet) {
   // Once a change is registered, the ticking count-down starts for the processQueue.
   if (this.changeQueue.length > 0 && !this.rendering_queued) {
     this.rendering_queued = true;
-    FastForEach.animateFrame.call(window, function () { self.processQueue(); });
+    if (isInitial)
```
Comment (brianmhunt) [inline index.js]: For consistency of style, the conditional bits should have the `{` and `}` :)

**Comment by brianmhunt** [comment]
That looks to be it.

The case where this is a downside is when there's a large bit of data to be initialized and the DOM has reflows queued.  I think that case is marginal, so I'd be ok with this PR.  If this becomes an issue we can add an option "delayInitialFrame" or some such.

To avoid regressions in future patches, do you have time to add one or two tests:
1. ensure that it's synchronous on the initial;
2. it's asynchronous otherwise (if this test isn't already there -- cannot recall offhand).

**Comment by cervengoc** [comment]
I was also thinking about really large data sets, but that case could be addressed by my other branch which implements partitioned processing. 

When possibly merging this two, I could image that the initial partition would be synchronous, and the others asynchronous. It would be a quite special case and of course after `applyBindings` one would not have a ready DOM again, but I think this all is an acceptable limitation (if we call it limitation at all).

By the way, if you have time, could you have a look at that mentioned branch too?

**Comment by cervengoc** [comment]
@brianmhunt Please check the changes

**Comment by brianmhunt** [comment]
Awesome.  Looks good to me.  Are you ok merging, adding to the change log, and bumping the version?  I can push update to npm.

**Comment by cervengoc** [comment]
I can merge I guess. I have still quite little git experience.

However, I would wait a bit with a new release, as this is not a huge issue I think, and there are some more things I'm working on, like the `parentNode` issue or the partitioning thing.

**Comment by brianmhunt** [comment]
Fair enough.  Thanks & keep me posted! :)

**Pull request merged**